src/main/java/randoop/generation/GrtFuzzing.java

```diff
@@ -0,0 +1,480 @@
+    for (int i = 0; i < numOfParams; i++) {
+      inputIndices.add(firstIndex + i);
+    }
+    return inputIndices;
+  }
+
+  /**
+   * Create a statement representing a Gaussian sampling result and append it to the given sequence.
+   *
+   * @param sequence the sequence to append the Gaussian sampling result to
+   * @param cls the class of the Gaussian number to be generated and appended
+   * @return a sequence with the Gaussian sampling result appended at the end
+   */
+  private static Sequence appendGaussianSampleSequence(Sequence sequence, Class<?> cls) {
+    Object gaussianSample = generateGaussianSample(cls);
+    Sequence gaussianSampleSequence = Sequence.createSequenceForPrimitive(gaussianSample);
+    return Sequence.concatenate(sequence, gaussianSampleSequence);
+  }
+
+  /**
+   * Generate a random Gaussian sample value N(0, GAUSSIAN_STD) of the given class.
+   *
+   * @param cls the class of the number to be generated
+   * @return a Gaussian sample value with 0 mean and a predefined standard deviation
+   */
+  private static Object generateGaussianSample(Class<?> cls) {
+    double randomGaussian = GAUSSIAN_STD * Randomness.nextRandomGaussian(1);
+    if (cls == int.class || cls == Integer.class) {
+      return (int) Math.round(randomGaussian);
+    } else if (cls == short.class || cls == Short.class) {
+      return (short) Math.round(randomGaussian);
+    } else if (cls == long.class || cls == Long.class) {
+      return Math.round(randomGaussian);
+    } else if (cls == float.class || cls == Float.class) {
+      return (float) randomGaussian;
+    } else if (cls == double.class || cls == Double.class) {
+      return randomGaussian;
+    } else {
+      throw new RuntimeException("Unexpected primitive type: " + cls.getName());
+    }
+  }
+
+  /**
+   * Get the methods for fuzzing primitive numbers of the given class.
+   *
+   * @param cls a numeric class
+   * @return a list of methods that will, together, be used to fuzz a number of the given class
+   */
+  private static List<Executable> getNumberFuzzingMethods(Class<?> cls)
+      throws NoSuchMethodException {
+    List<Executable> methodList = new ArrayList<>();
+
+    if (cls == int.class || cls == Integer.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (cls == double.class || cls == Double.class) {
+      methodList.add(Double.class.getMethod("sum", double.class, double.class));
+    } else if (cls == float.class || cls == Float.class) {
+      methodList.add(Float.class.getMethod("sum", float.class, float.class));
+    } else if (cls == long.class || cls == Long.class) {
+      methodList.add(Long.class.getMethod("sum", long.class, long.class));
+    } else if (cls == short.class || cls == Short.class) {
+      // Short doesn't have a sum method, so we use Integer.sum and get the short value
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
```

Comment:
Could the client just use `+` and a cast, rather than method calls?

Reply:
Added as TODO